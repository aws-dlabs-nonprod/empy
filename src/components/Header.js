import React, { Component } from 'react';
import styled from 'styled-components';
import Media from '@style/media';
import { connect } from 'react-redux';
import * as ActionTypes from '@constants/ActionTypes';
import { SoulMachinesContext } from '@contexts/SoulMachines';
import svgSprite from '@style/svg';
import CameraPreview from '@components/CameraPreview';

const logo = require('@svg/logo.svg');

class Header extends Component {

    static contextType = SoulMachinesContext;

    constructor() {
        super();


        this.handleGoHome = this.handleGoHome.bind(this);
    }

    handleGoHome() {
        const { isConnected } = this.props;

        if (isConnected) {
            this.props.showFeedback();
        } else {
            this.props.goHome();
        }
    }

    render() {
        const {
            isConnected,
            isTranscriptOpen,
            isMuted,
            hasCamera,
            hasMicrophone,
            isVideoCollapsed,
            isUserSpeaking
        } = this.props;

        return (
            <StyledHeader { ...this.props }>
                <div className="logo">
                    <div className="logo__image"
                        onClick={ this.handleGoHome }
                        dangerouslySetInnerHTML={{ __html: svgSprite(logo)}}>
                    </div>
                </div>
    
                { isConnected &&
                    <CameraPreview
                        isMuted={ isMuted }
                        hasCamera={ hasCamera }
                        hasMicrophone={ hasMicrophone }
                        isTranscriptOpen={ isTranscriptOpen }
                        isVideoCollapsed={ isVideoCollapsed }
                        isUserSpeaking={ isUserSpeaking }
                        toggleVideoCollapse={() => {
                            this.props.toggleVideoCollapse();
                        }}
                    />
                }
            </StyledHeader>
        );
    }
}

const StyledHeader = styled.header`
    left: 0%;
    padding: 1.5rem;
    position: fixed;
    right: 0;
    top: 0;
    height: 4.8rem;
    transform: translate3d(0, 0, 0);
    z-index: 4;

    ${Media.desktop`
        padding: 3rem;
        background: none;
        border: none;
        height: auto;
    `}

    .logo {
        display: flex;
        align-items: center;
    }

    .logo__image {
        cursor: pointer;
        width: 5.2rem;
        height: 2.1rem;
        margin: 0.7rem 1.5rem 0.7rem 0;

        ${Media.desktop`
            width: 6.9rem;
            height: 2.8rem;
            margin: 0 2rem 0 0;

            svg {
                width: 6.9rem;
                height: 2.8rem;
            }
        `}

        svg {
            display: block;
            width: 100%;
        }
    }
`;

function mapStateToProps(state) {
    return {
        isTranscriptOpen: state.isTranscriptOpen,
        isConnected: state.isConnected,
        isMuted: state.isMuted,
        hasCamera: state.hasCamera && state.isVideoEnabled,
        hasMicrophone: state.hasMicrophone,
        isVideoCollapsed: state.isVideoCollapsed,
        isUserSpeaking: state.isUserSpeaking
    };
}

function mapDispatchToProps(dispatch) {
    return {
        toggleVideoCollapse: () => dispatch({
            type: ActionTypes.TOGGLE_VIDEO_COLLAPSE
        }),

        goHome: () => dispatch({
            type: ActionTypes.SCENE_DISCONNECTED
        }),

        showFeedback: () => dispatch({
            type: ActionTypes.SHOW_FEEDBACK
        })
    };
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Header);
