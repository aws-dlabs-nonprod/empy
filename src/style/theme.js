export default {
    textPrimary: '#2D373E',
    textSecondary: '#E7E6E6',
    textBody: '#1A1A1A',
    textDark: '#621A4B',

    colourPrimary: '#D5002B',
    colourPrimaryHover: '#e24d6b',
    colourPrimaryTransparent: 'rgba(0, 174, 236, 0.5)',

    colourForeground: '#FFFFFF',
    colourForegroundHover: '#F9F9F8',
    colourForegroundTransparent: 'rgba(255, 255, 255, 0.2)',

    colourInactive: '#7F7F7F',
    colourBackground: '#E5E5E5',
    colourDivider: '#D7D2CB',
    colourFaintBorder: '#8C9296',

    radius: '0.3rem',
    splitHeight: '50vh'
};
