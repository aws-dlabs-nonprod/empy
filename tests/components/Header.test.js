import React from 'react';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import theme from '@style/theme';

import Header from '@components/Header';

const mockStore = configureStore([]);

// Mock getUserMedia
global.navigator.mediaDevices = {
    getUserMedia: () => {
        return new Promise(resolve => {});
    }
};

describe('Header', () => {

    test('renders correctly', () => {
        const store = mockStore({});

        const component = mount(
            <Provider store={store}>
                <ThemeProvider theme={ theme }>
                    <Header />
                </ThemeProvider>
            </Provider>
        );

        expect(component.find(Header)).toMatchSnapshot();
    });

});
