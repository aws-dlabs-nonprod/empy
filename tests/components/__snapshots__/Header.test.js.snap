// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c0 {
  left: 0%;
  padding: 1.5rem;
  position: fixed;
  right: 0;
  top: 0;
  height: 4.8rem;
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
  z-index: 4;
}

.c0 .logo {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .logo__image {
  cursor: pointer;
  width: 5.2rem;
  height: 2.1rem;
  margin: 0.7rem 1.5rem 0.7rem 0;
}

.c0 .logo__image svg {
  display: block;
  width: 100%;
}

@media (min-width:992px) and (min-height:0px) {
  .c0 {
    padding: 3rem;
    background: none;
    border: none;
    height: auto;
  }
}

@media (min-width:992px) and (min-height:0px) {
  .c0 .logo__image {
    width: 6.9rem;
    height: 2.8rem;
    margin: 0 2rem 0 0;
  }
}

<Connect(Header)>
  <Header
    goHome={[Function]}
    toggleVideoCollapse={[Function]}
  >
    <Header__StyledHeader
      goHome={[Function]}
      toggleVideoCollapse={[Function]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "Header__StyledHeader-sc-18ooop5-0",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "left:0%;padding:1.5rem;position:fixed;right:0;top:0;height:4.8rem;transform:translate3d(0,0,0);z-index:4;",
                "@media (min-width:",
                "992",
                "px) and (min-height:",
                "0",
                "px){",
                "
        padding: 3rem;
        background: none;
        border: none;
        height: auto;
    ",
                "}",
                " .logo{display:flex;align-items:center;}.logo__image{cursor:pointer;width:5.2rem;height:2.1rem;margin:0.7rem 1.5rem 0.7rem 0;",
                "@media (min-width:",
                "992",
                "px) and (min-height:",
                "0",
                "px){",
                "
            width: 6.9rem;
            height: 2.8rem;
            margin: 0 2rem 0 0;
        ",
                "}",
                " svg{display:block;width:100%;}}",
              ],
            },
            "displayName": "Header__StyledHeader",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "Header__StyledHeader-sc-18ooop5-0",
            "target": "header",
            "toString": [Function],
            "usesTheme": false,
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        goHome={[Function]}
        toggleVideoCollapse={[Function]}
      >
        <header
          className="c0"
        >
          <div
            className="logo"
          >
            <ReactSVG
              className="logo__image"
              evalScripts="never"
              fallback={null}
              loading={null}
              onClick={[Function]}
              onInjected={[Function]}
              renumerateIRIElements={true}
              src="mock-file"
              svgClassName={null}
              svgStyle={Object {}}
              wrapper="div"
            >
              <div
                className="logo__image"
                onClick={[Function]}
              />
            </ReactSVG>
          </div>
        </header>
      </StyledComponent>
    </Header__StyledHeader>
  </Header>
</Connect(Header)>
`;
